vendor/bat-native-ads/src/bat/ads/internal/eligible_ads/new_tab_page_ads/eligible_new_tab_page_ads_v2.cc

```diff
@@ -38,20 +38,23 @@ void EligibleAdsV2::GetForUserModel(
   BLOG(1, "Get eligible new tab page ads:");
 
   database::table::AdEvents database_table;
-  database_table.GetAll([=](const bool success, const AdEventList& ad_events) {
-    if (!success) {
-      BLOG(1, "Failed to get ad events");
-      callback(/* had_opportunity */ false, {});
-      return;
-    }
-
-    const int max_count = features::GetBrowsingHistoryMaxCount();
-    const int days_ago = features::GetBrowsingHistoryDaysAgo();
-    AdsClientHelper::Get()->GetBrowsingHistory(
-        max_count, days_ago, [=](const BrowsingHistoryList& browsing_history) {
-          GetEligibleAds(user_model, ad_events, browsing_history, callback);
-        });
-  });
+  database_table.GetForType(
+      mojom::AdType::kNewTabPageAd,
```

Comment:
We now only get a list of relevant ad types, and then calculate the eligible ads for that type. Previously we got all ad events and filtered other ad types within the frequency caps